hudi-common/src/main/java/org/apache/hudi/common/table/timeline/CompletionTimeQueryView.java

```diff
@@ -159,6 +166,50 @@ public Option<String> getCompletionTime(String startTime) {
       // the instant is still pending
       return Option.empty();
     }
+    loadCompletionTimeIncrementally(startTime);
+    return Option.ofNullable(this.startToCompletionInstantTimeMap.get(startTime));
+  }
+
+  /**
+   * Queries the instant start time with given completion time range.
+   *
+   * <p>By default, assumes there is at most 1 day time of duration for an instant to accelerate the queries.
+   *
+   * @param startCompletionTime The start completion time.
+   * @param endCompletionTime   The end completion time.
+   *
+   * @return The instant time set.
+   */
+  public Set<String> getStartTimeSet(String startCompletionTime, String endCompletionTime) {
+    // assumes any instant/transaction lasts at most 1 day to optimize the query efficiency.
```

Comment:
Typically it's a safe assumption as most commits last from a few minutes to an hour or so. But in some cases, when pipeline is blocked, commit can remain pending for longer duration. Should this be an internal config?

Reply:
Good point, I prefer we maintain it as an internal config first, we can expose it as public when receiving enough feedback from the community users.